library/public/image.h

```diff
@@ -0,0 +1,52 @@
+/**
+ * @class   image
+ * @brief   Class used to represent an image
+ *
+ * TODO improve doc
+ */
+
+#ifndef f3d_image_h
+#define f3d_image_h
+
+#include "export.h"
+
+#include <string>
+
+namespace f3d
+{
+class F3D_EXPORT image
+{
+public:
+  image();
+  explicit image(const std::string& path);
+
+  ~image();
+
+  image(const image& img);
+  image& operator=(const image& img);
+
+  image(image&& img);
+  image& operator=(image&& img);
+
+  int getWidth() const;
+  int getHeight() const;
+
+  image& setResolution(int width, int height);
+
+  int getChannelCount() const;
+  image& setChannelCount(int dim);
+
+  image& setData(unsigned char* buffer);
+  unsigned char* getData() const;
+
+  bool compare(const image& reference, image& result, double threshold) const;
+
+  image& save(const std::string& path);
```

Comment:
why do many of these methods return a ref to image ?

Reply:
I like this design, and I'd like to push more toward it. It's better in my opinion to return the object to chain calls like: 
 
```cpp
// nice
img.setResolution(300,300)
      .setChannelCount(3)
      .setData(buffer)
      .save("/tmp/img.png");

// not nice :(
img.setResolution(300,300)
img.setChannelCount(3)
img.setData(buffer)
img.save("/tmp/img.png");
```

Reply:
it makes sense with class that "are things" like image here, much less for class that "do things" like loader. My initial design was probably VTK-influenced hence the current API. I'm not against having a bit of both but I hope this will not be confusing for users. wdyt ?

Reply:
Yes you're right, it makes sense here, but much less for the loader. For the `options` class, it could be useful though to chain `set` calls.

Reply:
ok, feel free to improve !